tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/error_match_pattern_negative.bal

```diff
@@ -49,10 +57,74 @@ function testNegative2() {
 }
 
 function testNegative3() {
-    error v = getCustomError();
-    match v {
-        var errorA if v is MyError1 => { } // unused variable 'errorA'
-        var errorB if v is MyError2 => { } // unused variable 'errorB'
+    error v1 = getCustomError();
+    match v1 {
+        var errorA if v1 is MyError1 => { } // unused variable 'errorA'
+        var errorB if v1 is MyError2 => { } // unused variable 'errorB'
+    }
+
+    error? v2 = getCustomError();
+    match v2 {
+        var errorC if v2 is MyError1 => { } // unused variable 'errorC'
+        var errorD if v2 is MyError2 => { } // unused variable 'errorD'
+    }
+}
+
+function testNegative4() {
+    error v1 = getCustomError();
+    match v1 {
+        var errorA if v1 is MyError3 => { } // unused variable 'errorA'
+                                            // unnecessary condition: expression will always evaluate to 'true'
+        var errorB if v1 is MyError4 => { } // unused variable 'errorB', unreachable pattern
+                                            // unnecessary condition: expression will always evaluate to 'true'
+    }
+
+    error? v2 = getCustomError();
+    match v2 {
+        var errorC if v2 is MyError3 => { } // unused variable 'errorC'
+        var errorD if v2 is MyError4 => { } // unused variable 'errorD', unreachable pattern
```

Comment:
This `unreachable pattern` warning is not correct, right?

Reply:
It is correct.
Because MyError3 and MyError1 are not `distinct error` type.
Ballerina gives `error<record { int x; }> is error` as true.

```
public function main() {
    error<record { int x; }>? a = error("", x = 2);
    io:println(a is error); // print true
}
```

Reply:
This is my mistake, This is valid